JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -182,24 +142,50 @@ fun HomeScreen(
     ) { innerPadding ->
         val modifier = Modifier.padding(innerPadding)
         LoadingContent(
-            empty = posts.initialLoad,
+            empty = uiState.initialLoad,
             emptyContent = { FullScreenLoading() },
-            loading = posts.loading,
+            loading = uiState.loading,
             onRefresh = onRefreshPosts,
             content = {
                 HomeScreenErrorAndContent(
-                    posts = posts,
+                    posts = uiState.posts,
+                    isShowingErrors = uiState.errorMessages.isNotEmpty(),
                     onRefresh = {
                         onRefreshPosts()
                     },
                     navigateToArticle = navigateToArticle,
-                    favorites = favorites,
+                    favorites = uiState.favorites,
                     onToggleFavorite = onToggleFavorite,
                     modifier = modifier.supportWideScreen()
                 )
             }
         )
     }
+
+    // Process one error message at a time and show them as Snackbars in the UI
+    if (uiState.errorMessages.isNotEmpty()) {
+        val errorMessage = remember(uiState) { uiState.errorMessages[0] }
+
+        val errorMessageText: String = stringResource(errorMessage.messageId)
+        val retryMessage = stringResource(id = R.string.retry)
+
+        // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+        // don't restart the effect and use the latest lambda values.
+        val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+
+        LaunchedEffect(errorMessage, errorMessageText, scaffoldState) {
+            val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
+                message = errorMessageText,
+                actionLabel = retryMessage
+            )
+            if (snackbarResult == SnackbarResult.ActionPerformed) {
+                onRefreshPostsState()
+            }
+            // Once the message is displayed, notify the ViewModel
```

Comment:
it's once it's dismissed, not displayed!

Reply:
Don't understand your comment 🙃

Reply:
You should only reset the state once the snackbar is dismissed/times out, not after it's displayed.

Reply:
`showSnackbar` is a suspend function that resumes when it's dismissed

Reply:
yeah I know I'm just saying that the comment could be more precise, nothing wrong with code XD